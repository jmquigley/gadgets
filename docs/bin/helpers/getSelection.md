<a name="getSelectionShim"></a>

## getSelectionShim()
DOM Traversal is not implemented in JSDOM
The best we can do is shim the functions

**Kind**: global function  
