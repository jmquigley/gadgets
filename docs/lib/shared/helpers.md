## Functions

<dl>
<dt><a href="#debug">debug(context, args)</a></dt>
<dd><p>A wrapper for the debug function.  It uses the package.json to enable or
disable debugging information in the library.  It also turns off
coverage checks for this function.</p>
</dd>
<dt><a href="#globalize">globalize(name, pkg, replace)</a> ⇒ <code>object</code></dt>
<dd><p>Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is alread in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.</p>
</dd>
</dl>

<a name="debug"></a>

## debug(context, args)
A wrapper for the debug function.  It uses the package.json to enable or
disable debugging information in the library.  It also turns off
coverage checks for this function.

**Kind**: global function  

| Param | Type | Description |
| --- | --- | --- |
| context | <code>string</code> | the name of the module where this debug request is made. |
| args | <code>Array.&lt;object&gt;</code> | the list of arguments passed to the the debug function. |

<a name="globalize"></a>

## globalize(name, pkg, replace) ⇒ <code>object</code>
Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is alread in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.

**Kind**: global function  
**Returns**: <code>object</code> - the global instance reference  

| Param | Type | Description |
| --- | --- | --- |
| name | <code>string</code> | the name of the variable to store globally |
| pkg | <code>object</code> | the object associated with the variable name |
| replace | <code>boolean</code> | if true, then it will always overwrite the current global (if it exists) |

