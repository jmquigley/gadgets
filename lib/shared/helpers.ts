"use strict";

const debug = require("debug")("helpers");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
