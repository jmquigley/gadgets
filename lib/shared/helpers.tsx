'use strict';

const debugEntry = require('debug');
const pkg = require('../../package.json');

/**
 * A wrapper for the debug function.  It uses the package.json to enable or
 * disable debugging information in the library.  It also turns off
 * coverage checks for this function.
 * @param context {string} the name of the module where this debug request
 * is made.
 * @param args {object[]} the list of arguments passed to the the debug
 * function.
 */
export function debug(context: string, ...args: any[]) {
	/* istanbul ignore if  */
	if (pkg.debug) {
		const debugFn = debugEntry(`Gadgets -> ${context}`);
		debugFn(...args);
	}
}

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is alread in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(name: string, pkgname: any, replace: boolean = false) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
