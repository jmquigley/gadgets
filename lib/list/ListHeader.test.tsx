'use strict';

import {cleanup, mockupEnv} from '../../test/helpers';
mockupEnv();

import test from 'ava';
import {mount} from 'enzyme';
import * as path from 'path';
import * as React from 'react';
import {getDefaultListHeaderProps, ListHeader} from './index';

test.after.always.cb(t => {
	cleanup(path.basename(__filename), t);
});

test('Test retrieval of ListHeader props object', t => {
	const props = getDefaultListHeaderProps();

	t.truthy(props);
});

test('Test the creation of a ListHeader control with simple title', t => {
	const ctl = mount(
		<ListHeader
			title="test title"
		/>
	);

	t.truthy(ctl);

	t.is(ctl.prop('title'), 'test title');
	t.false(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));

	t.is(ctl.find('.ui-list-header').length, 1);
	t.is(ctl.find('.listHeader').length, 1);
});

// TODO: test case for validating props object creator
// TODO: disabling the accordion item
// TODO: make the accordion item invisible
