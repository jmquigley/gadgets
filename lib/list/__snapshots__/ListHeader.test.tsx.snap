// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test retrieval of ListHeader props object 1`] = `
Object {
  "backgroundColor": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "contentEditable": false,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "hiddenLeftButton": false,
  "hiddenRightButton": false,
  "id": null,
  "layout": 5,
  "left": null,
  "leftButton": null,
  "location": "none",
  "minHeight": null,
  "minWidth": null,
  "noedit": false,
  "nohover": true,
  "noripple": false,
  "notheme": false,
  "notooltip": false,
  "obj": "ListHeader",
  "onBlur": [Function],
  "onChange": [Function],
  "onClick": [Function],
  "onDoubleClick": [Function],
  "onFocus": [Function],
  "onKeyDown": [Function],
  "onKeyPress": [Function],
  "onMouseOut": [Function],
  "onUpdate": [Function],
  "padding": null,
  "rightButton": null,
  "selected": false,
  "sizing": "normal",
  "stacked": false,
  "style": Object {},
  "testing": true,
  "theme": null,
  "title": "â€‹",
  "tooltip": null,
  "top": null,
  "useedit": false,
  "visible": true,
  "widget": null,
  "width": null,
  "xcss": null,
}
`;

exports[`Test the creation of a ListHeader control with simple title 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hiddenLeftButton={false}
  hiddenRightButton={false}
  id={null}
  layout={5}
  left={null}
  leftButton={null}
  location="none"
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={true}
  noripple={false}
  notheme={false}
  notooltip={false}
  obj="ListHeader"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onDoubleClick={[Function]}
  onError={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onMouseOut={[Function]}
  onUpdate={[Function]}
  padding={null}
  reset={false}
  rightButton={null}
  selected={false}
  sizing="normal"
  stacked={false}
  style={Object {}}
  testing={true}
  theme={null}
  title="test title"
  tooltip={null}
  top={null}
  useedit={false}
  visible={true}
  widget={null}
  width={null}
  xcss={null}
>
  <Item
    backgroundColor={null}
    borderColor={null}
    bottom={null}
    className="ui-item nohover"
    color={null}
    contentEditable={false}
    controlled={true}
    direction="right"
    disabled={false}
    err={null}
    errorMessage={null}
    focus={false}
    height={null}
    hiddenLeftButton={false}
    hiddenRightButton={false}
    id={null}
    layout={5}
    left={null}
    leftButton={null}
    location="none"
    minHeight={null}
    minWidth={null}
    noedit={false}
    nohover={true}
    noripple={false}
    notheme={false}
    notooltip={false}
    obj="ListHeader"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onDoubleClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onMouseOut={[Function]}
    onUpdate={[Function]}
    padding={null}
    rightButton={null}
    selected={false}
    sizing="normal"
    stacked={false}
    style={Object {}}
    testing={true}
    theme={null}
    title="test title"
    tooltip={null}
    top={null}
    useedit={false}
    visible={true}
    widget={null}
    width={null}
    xcss={null}
  />
</Wrapper>
`;
